pkg/forwarder/forwarder.go

```diff
@@ -551,8 +568,15 @@ func (f *DefaultForwarder) SubmitV1CheckRuns(payload Payloads, extra http.Header
 }
 
 // SubmitV1Intake will send payloads to the universal `/intake/` endpoint used by Agent v.5
-func (f *DefaultForwarder) SubmitV1Intake(payload Payloads, extra http.Header, priority TransactionPriority) error {
-	transactions := f.createPriorityHTTPTransactions(v1IntakeEndpoint, payload, true, extra, priority)
+func (f *DefaultForwarder) SubmitV1Intake(payload Payloads, extra http.Header) error {
+	return f.submitV1Intake(payload, extra, f.createHTTPTransactions)
+}
+
+func (f *DefaultForwarder) submitV1Intake(
+	payload Payloads,
+	extra http.Header,
+	createHTTPTransactions func(endpoint endpoint, payload Payloads, apiKeyInQueryString bool, extra http.Header) []*HTTPTransaction) error {
+	transactions := createHTTPTransactions(v1IntakeEndpoint, payload, true, extra)
```

Comment:
So, maybe I'm missing something here.... Originally `SubmitMetadata()` and `SubmitHostMetadata()` were submitting to `metadataEndpoint`, but with this change we now use `submitV1Intake(...)` which will  create the transaction with destination endpoint  `v1IntakeEndpoint`. Now, I'm not sure what the current spec is for that endpoint but is this change intended?

Reply:
I think these updates come from https://github.com/DataDog/datadog-agent/pull/7077.

> Note: SubmitMetadata and SubmitHostMetadata were unused.
Before, Serializer.SendMetadata and Serializer.SendHostMetadata called Forwarder.SubmitV1Intake.
Now these 2 functions call respectively Forwarder.SubmitMetadata and Forwarder.SubmitHostMetadata which perform the same as Forwarder.SubmitV1Intake (Forwarder.SubmitHostMetadata use TransactionPriorityHigh)


Note: My target branch was originally `olivierg/simplify-forwarder-interface`.
Let me know if anything is unclear.